### PR TITLE
[Add Keyboard] Pancake v2

### PR DESCRIPTION
**Pull request by Spaceman**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Spaceman** [comment]
I'm not sure that the lint is important to pass here because it is failing due to the 'promicro' and 'feather' versions not having readme

File: keyboards/spaceman/pancake/rev2/rev2.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rev2.h"
```
Comment (drashna) [inline keyboards/spaceman/pancake/rev2/rev2.c:16]:
```suggestion
#include "rev2.h"


#ifdef OLED_DRIVER_ENABLE
__attribute__((weak)) oled_rotation_t oled_init_user(oled_rotation_t rotation) {
    return OLED_ROTATION_270;
}
__attribute__((weak)) void oled_task_user(void) {
    static const char PROGMEM pancake_logo[] = {
        0x00, 0x00, 0x3e, 0x0a, 0x04, 0x00, 0x3c, 0x0a, 0x3c, 0x00, 0x3e, 0x0c, 0x18, 0x3e, 0x00, 0x3e,
        0x22, 0x22, 0x00, 0x3c, 0x0a, 0x3c, 0x00, 0x3e, 0x08, 0x36, 0x00, 0x3e, 0x2a, 0x22, 0x00, 0x00,
        0x00, 0x30, 0xc8, 0x84, 0x84, 0x02, 0x02, 0x02, 0x02, 0x01, 0x01, 0x01, 0x01, 0x01, 0x19, 0x1d,
        0x1d, 0x0d, 0x01, 0x01, 0x01, 0x01, 0x01, 0x02, 0x02, 0x02, 0x02, 0x84, 0x84, 0xc8, 0x30, 0x00,
        0x00, 0x63, 0x94, 0x08, 0x08, 0x11, 0x71, 0x17, 0x13, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x62, 
        0xe2, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x11, 0x11, 0x11, 0x31, 0x08, 0x08, 0x94, 0x63, 0x00,
        0x00, 0x00, 0x03, 0x01, 0x01, 0x02, 0x02, 0x02, 0x02, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04,
        0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x06, 0x02, 0x02, 0x02, 0x01, 0x01, 0x00, 0x00, 0x00 };
    oled_write_raw_P(pancake_logo, sizeof(pancake_logo));
    // Host Keyboard Layer Status
    oled_set_cursor(0, 4);
    oled_write_P(PSTR("\nLAYER\n-----\n"), false);

    switch (get_highest_layer(layer_state)) {
        case _DEFAULT:
            oled_write_P(PSTR("DEFLT\n"), false);
            break;
        case _FN:
            oled_write_P(PSTR("FUNCT\n"), false);
            break;
        case _RAISE:
            oled_write_P(PSTR("RAISE\n"), false);
            break;
        default:
            // Or use the write_ln shortcut over adding '\n' to the end of your string
            oled_write_ln_P(PSTR("Undefined"), false);
            break;
    }
}
#endif
```

File: keyboards/spaceman/pancake/rev2/rules.mk [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/spaceman/pancake/rev2/rules.mk:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by Spaceman** [comment]
Sorry about that first time doing OLED code and wasn't sure where to put it.

**Comment by Spaceman** [comment]
Why did it remove qmk/collaborators? WTH

File: keyboards/spaceman/pancake/rev2/readme.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+Ortho 40% Ortho Keyboard with an OLED Display
 
 Keyboard Maintainer: [Spaceman](https://github.com/Spaceman)
-Hardware Availability: [GB](https://geekhack.org/index.php?topic=101371.0)
+Hardware Availability: [MKUltra.Click](https://mkultra.click/pancake-2/)
 
 Make example for this keyboard (after setting up your build environment):
 
-    make spaceman/pancake:default
+    qmk compile -kb spaceman/pancake/rev2 -km default
```
Comment (fauxpark) [inline keyboards/spaceman/pancake/rev2/readme.md:10]:
```suggestion
    make spaceman/pancake/rev2:default
```

File: keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,71 @@
+#include QMK_KEYBOARD_H
+enum layers {
+  _DEFAULT,
+  _LOWER,
```
Comment (fauxpark) [inline keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c:4]: I don't see a `_LOWER` layer here.

File: keyboards/spaceman/pancake/rev2/info.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
+{
+  "keyboard_name": "Pancake 2",
+  "keyboard_folder": "pancake/rev2",
+  "url": "",
+  "maintainer": "Spaceman",
+  "width": 13,
+  "height": 4,
+  "layouts": {
+		"LAYOUT_ortho_4x12": {
+      "key_count": 48,
```
Comment (fauxpark) [inline keyboards/spaceman/pancake/rev2/info.json:10]:
```suggestion
```

File: keyboards/spaceman/pancake/rev1/readme.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+Ortho 40% Ortho Keyboard with an option for the adafruit feather
+
+Keyboard Maintainer: [Spaceman](https://github.com/Spaceman)
+Hardware Availability: [MKUltra.Click](https://mkultra.click/group-buy-pancake-keyboard-kit)
+
+Make example for this keyboard (after setting up your build environment):
+
+    qmk compile -kb spaceman/pancake/rev1/feather/ -km default
+    or
+    qmk compile -kb spaceman/pancake/rev1/promicro -km default
```
Comment (fauxpark) [inline keyboards/spaceman/pancake/rev1/readme.md:12]:
```suggestion
    make spaceman/pancake/rev1/feather:default
    make spaceman/pancake/rev1/promicro:default
```

File: keyboards/spaceman/pancake/rev1/readme.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+# Pancake
+
+Ortho 40% Ortho Keyboard with an option for the adafruit feather
+
+Keyboard Maintainer: [Spaceman](https://github.com/Spaceman)
+Hardware Availability: [MKUltra.Click](https://mkultra.click/group-buy-pancake-keyboard-kit)
```
Comment (fauxpark) [inline keyboards/spaceman/pancake/rev1/readme.md:6]:
```suggestion
* Keyboard Maintainer: [Spaceman](https://github.com/Spaceman)
* Hardware Availability: [MKUltra.Click](https://mkultra.click/group-buy-pancake-keyboard-kit)
```

File: keyboards/spaceman/pancake/rev2/readme.md [hunk 8]
```diff
@@ -1,12 +1,12 @@
-# Pancake
+# Pancake 2
 
-Ortho 40% Ortho Keyboard with an option for the adafruit feather
+Ortho 40% Ortho Keyboard with an OLED Display
 
 Keyboard Maintainer: [Spaceman](https://github.com/Spaceman)
```
Comment (fauxpark) [inline keyboards/spaceman/pancake/rev2/readme.md:5]:
```suggestion
* Keyboard Maintainer: [Spaceman](https://github.com/Spaceman)
```

File: keyboards/spaceman/pancake/rev2/readme.md [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
-Hardware Availability: [GB](https://geekhack.org/index.php?topic=101371.0)
+Hardware Availability: [MKUltra.Click](https://mkultra.click/pancake-2/)
```
Comment (fauxpark) [inline keyboards/spaceman/pancake/rev2/readme.md:6]:
```suggestion
* Hardware Availability: [MKUltra.Click](https://mkultra.click/pancake-2/)
```

File: keyboards/spaceman/pancake/rev2/rules.mk [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+MOUSEKEY_ENABLE  = no	# Mouse keys
+EXTRAKEY_ENABLE  = yes	# Audio control and System control
+CONSOLE_ENABLE  = yes	# Console for debug
+COMMAND_ENABLE  = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE  = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no # enable or disable the keyboard backlight
+AUDIO_ENABLE  = no # This can be enabled if a speaker is connected to the expansion port. Not compatible with RGBLIGHT below
+RGBLIGHT_ENABLE  = no # This can be enabled if a ws2812 strip is connected to the expansion port.
+OLED_DRIVER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/spaceman/pancake/rev2/rules.mk:20]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
OLED_DRIVER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c (hunk 4 above)
Comment (Spaceman) [reply keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c:4]: This is a artifact from an old version the keymap removing now.

File: keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,70 @@
+#include QMK_KEYBOARD_H
+enum layers {
+  _DEFAULT,
+  _RAISE,
+  _FN
+};
+
+#define LOWER MO(_LOWER)
```
Comment (fauxpark) [inline keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c:8]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c (hunk 11 above)
Comment (Spaceman) [reply keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c]: Thanks, I'm a little slow

**Review by fauxpark — APPROVED** [review]

File: keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,68 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/spaceman/pancake/rev2/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/spaceman/pancake/rev2/rev2.c [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    oled_write_P(PSTR("\nLAYER\n-----\n"), false);
+
+    switch (get_highest_layer(layer_state)) {
+        case _DEFAULT:
+            oled_write_P(PSTR("DEFLT\n"), false);
+            break;
+        case _FN:
+            oled_write_P(PSTR("FUNCT\n"), false);
+            break;
+        case _RAISE:
```
Comment (drashna) [inline keyboards/spaceman/pancake/rev2/rev2.c:45]:
```suggestion
        case 0:
            oled_write_P(PSTR("DEFLT\n"), false);
            break;
        case 2:
            oled_write_P(PSTR("FUNCT\n"), false);
            break;
        case 1:
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by Spaceman** [comment]
Build is a-okay

**Comment by drashna** [comment]
Thanks!

**Pull request merged**